core/arch/arm/kernel/thread_optee_smc.c

```diff
@@ -35,7 +35,7 @@ void thread_handle_fast_smc(struct thread_smc_args *args)
 	thread_check_canaries();
 
 	if (IS_ENABLED(CFG_NS_VIRTUALIZATION) &&
-	    virt_set_guest(args->a7)) {
+	    (virt_set_guest(args->a7) && args->a7 != HYP_CLNT_ID)) {
```

Comment:
could remove a pair of parentheses.
Maybe test `args->a7 != HYP_CLNT_ID` before calling `virt_set_guest()`.